platform/viewer/cypress/integration/common/OHIFDownloadSnapshotFile.spec.js

```diff
@@ -0,0 +1,188 @@
+      .should('contain.text', 'Download High Quality Image');
+    // Check input fields
+    cy.get('.file-info-container')
+      .should('contain.text', 'Image width (px)')
+      .should('contain.text', 'Image height (px)')
+      .and('contain.text', 'File name')
+      .and('contain.text', 'File type')
+      .and('contain.text', 'Show Annotations');
+    cy.get('.file-type')
+      .find('select')
+      .select('png')
+      .should('have.value', 'png')
+      .select('jpg')
+      .should('have.value', 'jpg');
+    // Check image preview
+    cy.get('.preview').should('contain.text', 'Image Preview');
+    cy.get('[data-cy="viewport-preview-img"]')
+      .should('have.attr', 'src')
+      .and('include', 'data:image');
+    // Check buttons
+    cy.get('[data-cy="cancel-btn"]')
+      .scrollIntoView()
+      .should('have.text', 'Cancel');
+    cy.get('[data-cy="download-btn"]')
+      .scrollIntoView()
+      .should('have.text', 'Download');
+  });
+
+  it('checks displayed information for Desktop experience', function() {
+    // Set Desktop resolution
+    cy.viewport(1750, 720);
+    // Visual comparison
+    cy.screenshot('Download Image Modal - Desktop experience');
+    cy.percyCanvasSnapshot('Download Image Modal - Desktop experience');
+    //Check if all elements are displayed
+    cy.get('.OHIFModal')
+      .as('downloadImageModal')
+      .should('contain.text', 'Download High Quality Image');
+    // Check input fields
+    cy.get('.file-info-container')
+      .should('contain.text', 'Image width (px)')
+      .should('contain.text', 'Image height (px)')
+      .and('contain.text', 'File name')
+      .and('contain.text', 'File type')
+      .and('contain.text', 'Show Annotations');
+    cy.get('.file-type')
+      .find('select')
+      .select('png')
+      .should('have.value', 'png')
+      .select('jpg')
+      .should('have.value', 'jpg');
+    // Check image preview
+    cy.get('.preview').should('contain.text', 'Image Preview');
+    cy.get('[data-cy="viewport-preview-img"]')
+      .should('have.attr', 'src')
+      .and('include', 'data:image');
+    // Check buttons
+    cy.get('[data-cy="cancel-btn"]')
+      .scrollIntoView()
+      .should('have.text', 'Cancel');
+    cy.get('[data-cy="download-btn"]')
+      .scrollIntoView()
+      .should('have.text', 'Download');
+  });
+
+  it('downloads image file', function() {
+    cy.get('.width')
+      .find('input')
+      .clear()
+      .type('300');
+    cy.get('#file-name')
+      .clear()
+      .type('new-filename');
+    cy.get('.file-type')
+      .find('select')
+      .select('png');
+
+    // TO-DO: Implement a way to not trigger the button function to open the download browser dialog.
+    // Suggestion of approach: https://github.com/cypress-io/cypress/issues/949
+    // Once we can block the download dialog we can click on Download button.
+    // cy.get('[data-cy="download-btn"]')
+    //   .scrollIntoView()
+    //   .click();
```

Comment:
Is this something @rodrigolabs can assist with?

Reply:
With @rodrigolabs 's help, now we are able to test the `Download` button.
We made the download url accessible for cypress and we also blocked the dialog event once we the `Download` button is clicked. Then we verify if the captured url contains the string`"blob:"`.

A better approach would be checking if this url's request returns 200, but due to [this cypress issue](https://github.com/cypress-io/cypress/issues/2918), it is not possible to implement it yet.